crates/ursa-network/src/service.rs

```diff
@@ -491,19 +492,50 @@ where
         Ok(())
     }
 
-    fn handle_discovery(&mut self, discovery_event: DiscoveryEvent) -> Result<()> {
-        match discovery_event {
-            DiscoveryEvent::Connected(peer_id) => {
-                trace!("[DiscoveryEvent::Connected] - Peer connected {:?}", peer_id);
-                self.emit_event(NetworkEvent::PeerConnected(peer_id));
-            }
-            DiscoveryEvent::Disconnected(peer_id) => {
-                trace!(
-                    "[DiscoveryEvent::PeerDisconnected] - Peer disconnected {:?}",
-                    peer_id
-                );
-                self.emit_event(NetworkEvent::PeerDisconnected(peer_id));
+    pub fn handle_kad(&mut self, event: KademliaEvent) -> Result<()> {
+        match event {
+            KademliaEvent::OutboundQueryProgressed { id, result, .. } => match result {
```

Comment:
Can we handle these kad events

```
GetProviders
GetClosestPeers
```
This is where you can use the peer info

Reply:
> GetProviders

This wouldn't ever get generated as we're not using kad's providers or querying for them

> GetClosestPeers

In the current state this event is never generated (it's generated from `behaviour.kad.get_closest_peers()`), but I plan on using + handling it for random kad walks

Reply:
That make sense